### ISSUE TITLE
[Other - Accessibility] Helper text not programmatically associated with input. (00.00.1)

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#48249]
#### VFS team name
Benefits Team 1

#### VFS product name
Certificate of Eligibility

#### VFS feature name
Loan Guarantee: Certificate of Eligibility

#### Point of Contact/Reviewers
Brian DeConinck - @briandeconinck - Accessibility

*For more information on how to interpret this ticket, please refer to the [Anatomy of a Staging Review issue ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/Anatomy-of-a-Staging-Review-Issue-ticket.2060320997.html) guidance on Platform Website.

---

### Platform Issue
No VA.gov Experience Standard for the issue found.

### Issue Details
If a Veteran already has a VA-backed loan and is prompted to provide information about that loan in Step 4, the field "VA loan number" has helper text that reads "This number has 12 digits." However, that helper text isn't programmatically associated with the field or its label, and won't be announced by screen readers when a user tabs through inputs.

This almost falls under VA.gov Experience Standard 09.05 (Visual labels aren't included in the accessible label), which would be a launch-blocking issue. But since it's helper text and not the label itself, and it's not on a required field, that standard isn't a perfect fit for this issue. Logging it as an "Other" and non-launch-blocking instead.

### Link, screenshot or steps to recreate
Steps to reproduce:

1. Logged into staging, navigate through the COE request form to Step 4 of 6. 
2. For "Have you ever had a VA-backed loan?" answer Yes.
3. Answer "How will you use your Certificate of Eligibility?" (looks like any answer will lead to the same next step).
4. On the "Tell us about all your VA-backed loans" page, activate a screen reader (recommend VoiceOver on Mac or NVDA on Windows for testing).
5. Answer each question, pressing [tab] to move to the next input.
6. Observe that for "VA loan number" the helper text "This number has 12 digits" is not announced.

As currently coded, this question looks like this:

```
<label id="root_relevantPriorLoans_0_vaLoanNumber-label" class="schemaform-label" for="root_relevantPriorLoans_0_vaLoanNumber">VA loan number</label>
<div>This number has 12 digits.</div>
<div class="schemaform-widget-wrapper">
     <input type="text" id="root_relevantPriorLoans_0_vaLoanNumber" name="root_relevantPriorLoans_0_vaLoanNumber" class="coe-loan-input" value="">
</div>
```

The `label` references the `input` correctly, but the helper text is just wrapped in a `div` with no ID that's not referenced anywhere else.

### VA.gov Experience Standard
[Category Number 00, Issue Number 00](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/VA.gov-experience-standards.1683980311.html)

### Platform Recommendation
The helper or description text for this field should have its own unique ID (`<div id="some-unique-id">This number has 12 digits</div>`), and the `input` element should reference that ID using `aria-describedby`, eg. `<input ... aria-describedby="some-unique-id">`. (There are other ways of associating the helper text with the field, but this is the cleanest way.)

I'm not as familiar with the Forms Library as I should be, so this may already be supported in the form schema --- definitely worth checking the documentation.

If it's not supported and it's not possible to properly associate the helper text with the field, then I would recommend:
- Move the helper text to the `label`, so now the field's label reads "VA loan number (This number has 12 digits)" or something similar.
- Pass this ticket along to the Design System team to add to the Forms Library backlog.

---

### VFS Guidance

- Close the ticket when the issue has been resolved or validated by your Product Owner
- If your team has additional questions or needs Platform help validating the issue, please comment on the ticket
- Some feedback provided may be out of scope for your iteration of the product, however, Platform's OCTO leadership has stated that all identified issues need to be documented and it is still your responsibility to resolve the issue.
- If you do not believe that this Staging Review issue ticket is the responsibility of your team, comment below providing an explanation and who you believe is responsible. Please tag the Point of Contact/Reviewers. Governance team will research and will follow up.

**Comment by sortizsh** [comment]
Not a launch blocker, nice to have

**Comment by shiragoodman** [comment]
@sortizsh correct, this is not a launch-blocking issue, however it is also not a "nice to have". This is a Platform standard, and although you're not being asked to correct before launch, you will be required to fix in time.
